app/models/user.rb

```diff
@@ -5,9 +5,22 @@ class User < ActiveRecord::Base
          :recoverable, :rememberable, :trackable, :validatable
 
   has_many :lists
+  validates :api_key, uniqueness: true
+  before_create :assign_key
+
+  def assign_key
```

Comment:
Consider minimizing the visibility of `assign_key` and `generate_api_key` methods. It looks like they don't need to exposed as `public` methods on `User`, it'd be OK for them to be `private`.